### PR TITLE
Allow spaces in filenames in registry files

### LINKED ISSUE #313: Registry files cannot handle file names containing spaces
**Description of the problem:**

The way that `registry.txt` files are parsed [here](https://github.com/fatiando/pooch/blob/main/pooch/core.py#L659) does not allow filenames to contain spaces. This could be fixed by implementing advanced parsing that allows escaping using enclosing quotes or backslashes.

**Full code that generated the error**

With [registry.txt](https://github.com/fatiando/pooch/files/9130660/registry.txt) in path:

```python
import pooch
P = pooch.create(
    path=pooch.os_cache("test"),
    base_url="ignored"
)
P.load_registry("registry.txt")
```

**Full error message**

```
OSError: Invalid entry in Pooch registry file './registry.txt': expected 2 or 3 elements in line 1 but got 7. Offending entry: '"How To - Minimal Workflow.pdf" sha256:7d4da5fe4d0438437834fe846608b93959dcb80ded941eb85663b2d905c54000 https://heidata.uni-heidelberg.de/api/access/datafile/:persistentId?persistentId=doi:10.11588/data/TJNQZG/DWSXML'
```

### PR DESCRIPTION
**Pull request by dokempf**
Previously, the tokenization would not allow this. The commit introduces
`shlex.split` as a tokenizer that is aware of quotes and escapes.

This fixes #313

### CODE REVIEW

**Comment by dokempf** [comment]
I think this is ready for review. I had a look at the CodeCov failure and it seems quite unrelated, the changes on this PR should be all covered.

File: pooch/tests/data/tiny space data.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# A tiny data file for test purposes only
```
Comment (leouieda) [inline pooch/tests/data/tiny space data.txt:1]: Is this file needed? Loading the registry doesn't look for the files mentioned there.

File: pooch/tests/data/registry-spaces.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+"tiny space data.txt" baee0894dba14b12085eacb204284b97e362f4f3e5a5807693cc90ef415c1b2d
+tiny\ space\ data.txt baee0894dba14b12085eacb204284b97e362f4f3e5a5807693cc90ef415c1b2d
```
Comment (leouieda) [inline pooch/tests/data/registry-spaces.txt:2]: Maybe make this a different file name so we can check for its existence in the test? As it is, this will clobber the entry above.

**Review by leouieda — CHANGES_REQUESTED** [review]
Thanks @dokempf! A few minor questions and suggestions only. I don't know why codecov failed but maybe it will go away when the CI runs again.

File: pooch/tests/data/tiny space data.txt (hunk 1 above)
Comment (dokempf) [reply pooch/tests/data/tiny space data.txt]: Ah, I did not know that the files do not need to exist. That makes things easier, of course.

**Review by leouieda — APPROVED** [review]

**Comment by leouieda** [comment]
Thanks @dokempf! Looks great to me. Just waiting on CI to finish 🤖.

**Comment by leouieda** [comment]
Thanks for this @dokempf! I'll make a release soon with this and your other PR.

**Comment by dokempf** [comment]
@leouieda Great, I would try to implement #317 as well before that release.

**Comment by mscheltienne** [comment]
I'm running v1.7.0, and it seems like this is not fully resolved. 
Filename: "LICENSE (copy)", with a space.

Run: `pooch.make_registry(DATASET, output=REGISTRY, recursive=True)` with `DATASET` the path to the folder containing `LICENSE (copy)` and more. It results in this registry:

```
LICENSE 1f7e3edb3fc584df7f618e17cef4befa18232b99302dd8d55e519ab3d9e028b4
LICENSE (copy) 1f7e3edb3fc584df7f618e17cef4befa18232b99302dd8d55e519ab3d9e028b4
ssp/ssp_0_230120.fif b34bd4c97e8e32854c40dd01025bb5f236f6def3cf77fbb16b8e4c0f0c3e69d1
ssp/ssp_0_ias_230120.fif ba8d51427be963268884f6dc2b994b494268e1e80648aaf8cffca2a01240cb19
ssp/ssp_60_230120.fif 9465dcabfb9808b13eced54a644b9104f3d5d8b348bb4315f76dd572eff35ef2
ssp/ssp_60_ias_230120.fif 14f55a49ae8fba62c868fb117daf902f32981cecf7bc8675a4e21e44018d9ce8
ssp/ssp_68_230120.fif c6b5f56a30582b930e5c83f4404c82579929fde8cc650d062dc33429e95ebc4f
ssp/ssp_68_ias_230120.fif 379ea05dc48869c57368d354f17e65e943d555322d63b3b5caba3ae7221374cf
version.txt e2556a181068db2c7e3b2b127de33540448820fb1e97da29239833b6a8e09764
```

And now `fetcher.load_registry(REGISTRY)` with `fetcher` being an instance of Pooch yields:

```
fetcher.load_registry(REGISTRY)
fetcher.registry

>>>
{'LICENSE': '(copy)',
 'ssp/ssp_0_230120.fif': 'b34bd4c97e8e32854c40dd01025bb5f236f6def3cf77fbb16b8e4c0f0c3e69d1',
 'ssp/ssp_0_ias_230120.fif': 'ba8d51427be963268884f6dc2b994b494268e1e80648aaf8cffca2a01240cb19',
 'ssp/ssp_60_230120.fif': '9465dcabfb9808b13eced54a644b9104f3d5d8b348bb4315f76dd572eff35ef2',
 'ssp/ssp_60_ias_230120.fif': '14f55a49ae8fba62c868fb117daf902f32981cecf7bc8675a4e21e44018d9ce8',
 'ssp/ssp_68_230120.fif': 'c6b5f56a30582b930e5c83f4404c82579929fde8cc650d062dc33429e95ebc4f',
 'ssp/ssp_68_ias_230120.fif': '379ea05dc48869c57368d354f17e65e943d555322d63b3b5caba3ae7221374cf',
 'version.txt': 'e2556a181068db2c7e3b2b127de33540448820fb1e97da29239833b6a8e09764'}
```

Which is wrong.

**Comment by dokempf** [comment]
@mscheltienne You need to adapt your registry file to put `".."` around the filename that contains spaces or use `\` to escape the whitespace character. I guess there is no way around this limitation as the parsing would be ambiguous otherwise.

**Comment by mscheltienne** [comment]
I agree, but that should be done automatically by `pooch` since this registry file was created with `pooch.make_registry`.

**Comment by dokempf** [comment]
Oh, sorry for my poor reading skills. I agree, `pooch` should output a quoted filename in this case.
